db/post_migrate/20190519130537_remove_boosts_widening_audience.rb

```diff
@@ -0,0 +1,23 @@
+class RemoveBoostsWideningAudience < ActiveRecord::Migration[5.2]
+  disable_ddl_transaction!
+
+  def up
+    public_boosts = Status.find_by_sql(<<-SQL)
+      SELECT boost.id, boost.visibility, boost.reblog_of_id
+      FROM statuses AS boost
+      LEFT JOIN statuses AS boosted ON boost.reblog_of_id = boosted.id
+      WHERE
+        boost.id > 101746055577600000
+        AND (boost.local = TRUE OR boost.uri IS NULL)
+        AND boost.visibility IN (0, 1)
+        AND boost.reblog_of_id IS NOT NULL
+        AND boosted.visibility = 2
+    SQL
+
+    RemovalWorker.push_bulk(public_boosts.pluck(:id))
```

Comment:
`find_by_sql` already loads full models, it does not return a relation, so `pluck` is equivalent to `map(&:id)` here

Reply:
Hm, ok, parts of the Mastodon code do that though. Is there something lighter we could do? We only need the id.